src/CryptoWorker.ts

```diff
@@ -1,27 +1,38 @@
 import { expose } from "threads/worker"
 import * as crypto from 'crypto'
-import { Buffer } from 'buffer/'
+import { cryptoConstants, deconstructChunk } from "./util"
 
-let cipher: crypto.Cipher
-let decipher: crypto.Decipher
+function encryptBlock(blockBuffer: Buffer, masterKey: Buffer, algorithm: 'aes-128-gcm' | 'aes-192-gcm' | 'aes-256-gcm', salt: Buffer, initVector: Buffer): Buffer {
+  // Initialize cipher
+  const key = crypto.pbkdf2Sync(masterKey, salt, cryptoConstants.PBKDF_NUM_ITERATIONS, cryptoConstants.KEY_LEN, 'sha512')
+  const cipher = crypto.createCipheriv(algorithm, key, initVector)
 
-function init(algorithm: string, key: Buffer, initVector: Buffer): void {
-  cipher = crypto.createCipheriv(algorithm, key, initVector)
-  decipher = crypto.createDecipheriv(algorithm, key, initVector)
-}
+  // Encrypt the blockBuffer
+  const encrypted = Buffer.concat([cipher.update(blockBuffer), cipher.final()])
 
-function encryptBuf(plainBuf: Buffer): Buffer {
-  return Buffer.from(cipher.update(plainBuf))
-}
+  // Extract the auth tag
+  const tag = cipher.getAuthTag()
 
-function decryptBuf(encryptedBuf: Buffer): Buffer {
-  return Buffer.from(decipher.update(encryptedBuf))
+  // Construct chunk
+  return Buffer.concat([salt, initVector, tag, encrypted])
 }
 
-export {init, encryptBuf, decryptBuf}
+function decryptChunk(chunkBuffer: Buffer, masterKey: Buffer, algorithm: 'aes-128-gcm' | 'aes-192-gcm' | 'aes-256-gcm'): Buffer {
```

Comment:
Since the underlying impl is shared between `Crypto.ts` and `CryptoWorker.ts`, it's possible to abstract this into a shared `CryptoUtils`. Not necessary since its so small.

Reply:
I have just added it to the `Crypto.ts` and exported it for use in our web worker file